packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -258,6 +258,34 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
   /// ** See code in examples/api/lib/material/scaffold/scaffold_messenger_state.show_snack_bar.0.dart **
   /// {@end-tool}
   ScaffoldFeatureController<SnackBar, SnackBarClosedReason> showSnackBar(SnackBar snackBar) {
+    assert(() {
```

Comment:
This could be wrapped up in a `debug...` method for reuse.

Reply:
@Piinks How? These are two different classes (and one is deprecated), so I am not sure how that would work. I could create a function, but I think the old method will be removed anyway.

Do you prefer I create a function for it?

Reply:
It has happened before that something that was deprecated is reverted and kept. `debug.dart` is already imported here, that seems like the right place for it. Although, this is causing internal test failures like the last time this change was proposed. I think this approach inadvertently catches a valid call to these methods.

Reply:
@Piinks what is the internal failure? It does not seem valid to call it during build because `setState` below this assert will throw in the same way.

Reply:
I've checked in on this, and the internal failure is the same one as the earlier PR that tried to address this caught: https://github.com/flutter/flutter/pull/67144


The G testing shard here is green because the test run is a subset of G tests. I cloned this and ran a global test suite on it which resulted in the failure. I don't think we can land this change.

Reply:
@Piinks is it not possible to change the G code? It (at least seems so) seems clear the code cannot be proper usage.

Reply:
If it were improper use the G code would be crashing with the 'unintuitive' error message (#66789) that this PR is trying to improve. I think this is accidentally catching valid code, because it currently exists and runs without error.

Reply:
@Piinks So what is that valid code? Why can we not fix the assertion?

Reply:
What I mean to say is that this assertion that you are adding in this PR catches the error you intend to catch, but it is also catching other cases that are not valid to assert. 

While I cannot share internal code, it stands to reason that if the internal code were invalid, it would crash with the same  bad error message you are trying to fix here. I am going to close this PR as the internal test proves that it asserts when it should not. Unfortunately, I do not have a simple reproduction of this case. 

I appreciate your work on this, it just does not work in the way that you expect.